### PR TITLE
Refacto updateAchievment

### PR DESCRIPTION
**Pull request by YouriDK**
Refacto updateAchievment

### CODE REVIEW

File: server/src/utils/fight/updateAchievements.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-await-in-loop */
```
Comment (Zenoo) [inline server/src/utils/fight/updateAchievements.ts]: The goal here is to not have any await in the for loops anymore :D

```suggestion
```

File: server/src/utils/fight/updateAchievements.ts [hunk 2]
```diff
@@ -6,101 +7,46 @@ const updateAchievements = async (
   store: AchievementsStore,
   isTournamentFight: boolean,
 ) => {
+  const bruteAcheivments = [];
+  const bruteAcheivmentsTournamentRelated = [];
```
Comment (Zenoo) [inline server/src/utils/fight/updateAchievements.ts]:
```suggestion
  const bruteAchievments = [];
  const bruteAchievmentsTournamentRelated = [];
```

File: server/src/utils/fight/updateAchievements.ts [hunk 3]
```diff
@@ -6,101 +7,46 @@ const updateAchievements = async (
...
-            data: {
-              achievement: name,
-              achievementCount: count,
-              bruteId,
-              date: new Date(),
-            },
-            select: { id: true },
-          });
-        }
+      const achievementName = _name as AchievementName;
```
Comment (Zenoo) [inline server/src/utils/fight/updateAchievements.ts:13]: Same as the line above?

**Review by Zenoo — CHANGES_REQUESTED** [review]
The whole system checking whether the achievement should be added or used as a max is missing.
You can check the `max` field here to check if an achievement needs to be handled as a max instead of an addition.

image

File: server/src/utils/fight/updateAchievements.ts (hunk 2 above)
Comment (YouriDK) [reply server/src/utils/fight/updateAchievements.ts:11]: I don't see any difference in your suggestion ^^"

**Review by Zenoo — APPROVED** [review]
Thanks for the hard work !

**Pull request merged**